### DISCUSSION TITLE
How can I deploy static files after a build on Vercel?

### DISCUSSION THREAD

**Discussion by jayhale** [vercel/vercel#6128] [Help] [answered]
For a project, I want to build both a function and static files - how can I do this in the new runtime semantic?

I'm reworking [@ardnt/vercel-python-wsgi](https://github.com/ardnt/vercel-python-wsgi) into the new runtimes convention.  I remember reading somewhere about writing files to different `.vercel_.../` folders to indicate which are for the function and which are static, but I can't find that reference.

In the past, I would define two builds - one using the WSGI builder and another using the static builder, e.g.:
```json
  "builds": [
    {
      "src": "now_build_staticfiles.sh",
      "use": "@now/static-build",
      "config": {
        "distDir": "staticfiles_build"
      }
    },
    {
      "src": "app/wsgi.py",
      "use": "@ardnt/now-python-wsgi",
      "config": {
        "maxLambdaSize": "30mb"
      }
    }
  ],
```

I see this documentation on frameworks that hints at a solution - is this the right path? https://vercel.com/docs/more/adding-your-framework

**Comment by styfle** [comment, marked as answer]
Hi James,

The [builds](https://vercel.com/docs/configuration#project/builds) property is a legacy feature [@now/static-build](http://npmjs.com/package/@now/static-build) is deprecated.

The document you linked to is indeed the correct way to add a framework (previously you needed custom builders).

However, Vercel already supports WSGI Python apps with zero configuration. https://vercel.com/docs/runtimes#official-runtimes/python

For static files, you can specify a build command in your project settings. https://vercel.com/docs/build-step

**Reply by jayhale** [reply]
Thanks - do I need to create two projects with two different build commands?

1. Build and deploy the function (e.g., `pip install -r requirements.txt`)
2. Build and deploy the static files (e.g., `pip install -r requirements.txt && python manage.py buildstatic`)

It's unclear to me how I can generate a function and static files (to be served via CDN) in the same project.  I only see one option for setting a build command.

---

Thanks for pointing to the existing builder - I'd prefer to use our custom builder / runtime though given it avoids having an opinion on the source tree.

**Reply by styfle** [reply]
> do I need to create two projects with two different build commands?

No one project will do.

The build command is used to generate outputs to the `public` directory. These are your static files.

Then the `/api` directory will look for files ending in `.py` and build those as Serverless Functions.

**Reply by jayhale** [reply]
Okay, I see, if I understand right:

- Runtimes build functions (e.g., the python runtime)
- The build command, from the project portal, builds static files

Will see if I can get things working again. Thanks!
